### DISCUSSION TITLE
I don’t suppose this format can be made to also work with MacOS on ARM (or other ARM)?

### DISCUSSION THREAD

**Discussion by gisborne** [jart/cosmopolitan#176] [Ideas]
The answer is probably no, but it would obviously be much more future-proof if this can be made to work on Macs using ARM.

**Comment by gisborne** [comment]
I get that this would make the executable bigger because it would need to include separate ARM code. Still…

**Reply by jart** [reply]
It would make the executable bigger. It's something that's possible and is likely to happen eventually. How soon I can't say. I talk about it at some length in this blog post https://justine.lol/ape.html

**Reply by mskyttner** [reply]
With some devops toil and a fat container runtime it should run (after a bloated docker/podman install followed by a 10 Mb cosmopolitan image download)? But that perhaps defeats the spirit of this effort.

But it wouldn't run on a Mac with arm64 if the binary was extracted from the image to the host? Below docker is used only to download and extract the binary from a multiarch Alpine Linux image where redbean was built on linux/arm64 architecture, so if your host matches that arch in the manifest, you will get that image.

```bash
export cid=$(docker create ghcr.io/mskyttner/cosmopolitan)
docker cp $cid:/usr/local/bin/redbean.com .
docker rm -v $cid
chmod +x redbean.com
./redbean.com -p 8080 -vv
firefox http://localhost:8080/
```
Once copied out from the image/container, the executable runs directly on the host if using for example Linux Mint. I have no Mac to test it but if I understand the blog post, it still shouldn't work there, right? And I don't think the image yet contains an embedded arm build of the emulator that would need to be used, if that is one step towards what is eventually needed?

**Reply by jart** [reply]
I don't use Docker so I don't know. Cosmopolitan is for x86 machines.
